src/main/kotlin/io/emeraldpay/dshackle/upstream/grpc/GrpcUpstreams.kt

```diff
@@ -92,7 +94,14 @@ class GrpcUpstreams(
             chanelBuilder.usePlaintext()
         }
 
-        val client = ReactorBlockchainGrpc.newReactorStub(chanelBuilder.build())
+        val client = ReactorBlockchainGrpc.newReactorStub(chanelBuilder.build()).let {
+            var builder = it
+            if (compression) {
+                builder = builder.withCompression(Codec.Gzip().messageEncoding)
+                log.info("Compression enabled for gRPC client")
```

Comment:
too small amount of data ( There are a lot of different grpc clients created here.